pkgs/development/python-modules/datalad-next/default.nix

```diff
@@ -0,0 +1,106 @@
+{
+  annexremote,
+  buildPythonPackage,
+  datalad,
+  datasalad,
+  fetchFromGitHub,
+  git,
+  git-annex,
+  humanize,
+  lib,
+  more-itertools,
+  psutil,
+  pythonOlder,
+  setuptools,
+  openssh,
+  unzip,
+  webdavclient3,
+}:
+
+buildPythonPackage rec {
+  pname = "datalad-next";
+  version = "1.5.0";
+  pyproject = true;
+  disabled = pythonOlder "3.12";
+
+  src = fetchFromGitHub {
+    owner = "datalad";
+    repo = "datalad-next";
+    rev = "refs/tags/${version}";
+    hash = "sha256-fqP6nG2ncDRg48kvlsmPjNBOzfQp9+7wTcGvsYVrRzA=";
+  };
+
+  nativeBuildInputs = [ git ];
+
+  build-system = [ setuptools ];
+
+  dependencies = [
+    annexremote
+    datasalad
+    datalad
+    humanize
+    more-itertools
+  ];
+
+  preCheck = ''
+    export HOME=$TMPDIR
+  '';
+
+  nativeCheckInputs = [
```

Comment:
Yes, but you forgot about this: 

> By default tests are run because [doCheck = true](https://nixos.org/manual/nixpkgs/unstable/#var-stdenv-doCheck). Test dependencies, like e.g. the test runner, should be added to [nativeCheckInputs](https://nixos.org/manual/nixpkgs/unstable/#var-stdenv-nativeCheckInputs).`

I am running your code build locally and it doesn't run any tests. When I add `pytestCheckHook` they fail.